File: bryntum/gantt/examples/php/php/Bryntum/Gantt/DependencySyncHandler.php
<?php

namespace Bryntum\Gantt;

class DependencySyncHandler extends \Bryntum\CRUD\SyncHandler
{

    private $gantt;

    public function __construct(&$gantt)
    {
        $this->gantt = &$gantt;
    }

    protected function prepareData(&$data)
    {
        // initialize response part related to the record
        $response = array();

        $taskIds = $this->gantt->phantomIdMap['tasks'];

        if (isset($data['type'])) {
            $data['typ'] = $data['type'];
            unset($data['type']);
        }

        // get newly created task Ids if these are references to phantom tasks
        if (isset($data['fromEvent']) && isset($taskIds[$data['fromEvent']])) {
            // use & return actual Id
            $data['fromEvent'] = $response['fromEvent'] = $taskIds[$data['fromEvent']];
        }

        if (isset($data['toEvent']) && isset($taskIds[$data['toEvent']])) {
            // use & return actual Id
            $data['toEvent'] = $response['toEvent'] = $taskIds[$data['toEvent']];
        }

        return $response;
    }

    public function add(&$dependency)
    {
        $response = $this->prepareData($dependency);
        $this->gantt->saveDependency($dependency);
        return $response;
    }

    public function update(&$dependency)
    {
        $response = $this->prepareData($dependency);
        $this->gantt->saveDependency($dependency);
        return $response;
    }

    public function remove($dependency)
    {
        $response = [];
        $this->gantt->removeDependency($dependency['id']);
        return $response;
    }
}

File: bryntum/gantt/examples/aspnetcore/BryntumGantt/Bryntum/Gantt/Request/Handler/DependencySyncHandler.cs
using Bryntum.CRUD.Request;
using Bryntum.CRUD.Response;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bryntum.Gantt.Request.Handler
{
    public class DependencySyncHandler : SyncStoreRequestHandler<Dependency> {

        private Gantt gantt;

        public DependencySyncHandler(Gantt gantt) {
            this.gantt = gantt;
        }

        public override Dependency GetEntity(IDictionary<String, Object> changes) {
            return gantt.GetDependency(Convert.ToInt32(changes["Id"]));
        }

        protected IDictionary<String, Object> PrepareData(Dependency dependency) {
            // initialize response part related to the record
            IDictionary<String, Object> response = new Dictionary<String, Object>();

            String phantomFrom = dependency.PhantomFromId;
            if (!String.IsNullOrEmpty(phantomFrom))
            {
                int? from = gantt.GetTaskIdByPhantom(phantomFrom);
                if (from.HasValue)
                {
                    dependency.FromIdRaw = from.Value;
                    // put updated From to response
                    response.Add("From", from);
                }
            }

            String phantomTo = dependency.PhantomToId;
            if (!String.IsNullOrEmpty(phantomTo))
            {
                int? to = gantt.GetTaskIdByPhantom(phantomTo);
                if (to.HasValue)
                {
                    dependency.ToIdRaw = to.Value;
                    // put updated To to response
                    response.Add("To", to);
                }
            }

            return response;
        }

        public override IDictionary<String, Object> Add(Dependency dependency)
        {
            IDictionary<String, Object> response = PrepareData(dependency);
            gantt.SaveDependency(dependency);
            return response;
        }

        public override IDictionary<String, Object> Update(Dependency dependency, IDictionary<String, Object> changes)
        {
            // apply changes to tthe entity
            if (changes.ContainsKey("Cls")) dependency.Cls = Convert.ToString(changes["Cls"]);
            if (changes.ContainsKey("From")) dependency.FromId = Convert.ToString(changes["From"]);
            if (changes.ContainsKey("To")) dependency.ToId = Convert.ToString(changes["To"]);
            if (changes.ContainsKey("Type")) dependency.Type = Convert.ToInt32(changes["Type"]);
            if (changes.ContainsKey("Lag")) dependency.Lag = Convert.ToDecimal(changes["Lag"]);
            if (changes.ContainsKey("LagUnit")) dependency.LagUnit = Convert.ToString(changes["LagUnit"]);
            if (changes.ContainsKey(dependency.PhantomIdField))
                dependency.PhantomId = Convert.ToString(changes[dependency.PhantomIdField]);

            IDictionary<String, Object> response = PrepareData(dependency);
            gantt.SaveDependency(dependency);
            return response;
        }

        public override IDictionary<String, Object> Remove(Dependency dependency)
        {
            IDictionary<String, Object> response = new Dictionary<String, Object>();
            gantt.RemoveDependency(dependency);
            return response;
        }
    }
}
